### ISSUE TITLE
No rest() and emplace()

### ISSUE THREAD

**Issue by devgs** [thelink2012/any#3]
Probably draft was updated since the original implementation. These member functions are very essential AFAIK.

Reference:
http://en.cppreference.com/w/cpp/utility/any

**Comment by thelink2012** [comment]
Yeah, see #2, still not entirely sure what we should do about this interface change.

**Comment by devgs** [comment]
Thanks for the prompt reply!

Are there any specific issues related to adding them? It seems like interface is settled now. At least for the last month or so I didn't notice any changes.

**Comment by thelink2012** [comment]
The major issue I see is with people that is using the old interface and blindly clones the repo without the proper commit hash. Plus, whether we should come up with a `#ifdef` to switch between interfaces.... well, that'll look a bit ugly, eh?

Yes, don't think the interface will change any further as the major compilers are implementing it outside of the experimental namespace already.

**Comment by devgs** [comment]
Are you okay with me cloning your repository and making adjustments?

**Comment by thelink2012** [comment]
Yes, I'm okay with that.
